app/models/group.rb

```diff
@@ -19,7 +19,21 @@ class Group < ActiveRecord::Base
   has_many :leaders, -> { where(group_members: { leader: true }) },
            through: :group_members, source: :user
 
+  after_destroy :destroy_notifications
+
   def led_by?(user)
     leaders.include? user
   end
+
+  def notifications
+    Notification.where('uniqueid ilike ?', '%new_group%').select do |n|
```

Comment:
It looks like we've been storing uniqueid as "new_group_#{current_user.id}", not "new_group_#{groupid}" (see services/group_notifier.rb:48) :(

Here is an example of an instance of Notification
```
#<Notification:0x007ff28744ad50
  id: 2,
  userid: 2,
  uniqueid: "new_group_member_4",
  data:
   "{\"user\":\"Karen\",\"groupid\":1,\"group\":\"Students with Anxiety\",\"type\":\"new_group_member\",\"uniqueid\":\"new_group_member_4\"}",
  created_at: Mon, 04 Jul 2016 16:30:28 UTC +00:00,
  updated_at: Mon, 04 Jul 2016 16:30:28 UTC +00:00>
```

I played around with it a little bit in the console and this works: 

```ruby
query_string = "groupid\":#{groupid},"
Notification.where('data ilike ?', "%#{query_string}%")
```

I don't like the `\":` piece, and it seems like it could potentially match data strings that it's not supposed to (the `,` is included at the end of the query string to ensure that `groupid: 1` doesn't also match `groupid: 11`, for example), although this does seem pretty unlikely considering the contents of the rest of the string. 

I think both solutions are about the same in terms of elegance, the former is more efficient while the latter is more rigorous.

However, I don't even think the latter allows us to achieve a group has_many :notifications association either because it still needs a foreign key.